src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -264,31 +264,67 @@ public static void Move(string sourceDirName, string destDirName)
 
             string fullsourceDirName = Path.GetFullPath(sourceDirName);
             string sourcePath = PathInternal.EnsureTrailingSeparator(fullsourceDirName);
+            string sourceDirectoryName = Path.GetFileName(fullsourceDirName);
 
             string fulldestDirName = Path.GetFullPath(destDirName);
             string destPath = PathInternal.EnsureTrailingSeparator(fulldestDirName);
+            string destDirectoryName = Path.GetFileName(fulldestDirName);
 
-            StringComparison pathComparison = PathInternal.StringComparison;
+            bool sameDirectoryIgnoreCase =
+                string.Equals(sourcePath, destPath, StringComparison.OrdinalIgnoreCase);
+            bool sameDirectoryNameIgnoreCase
+                = !string.IsNullOrEmpty(sourceDirectoryName) && string.Equals(sourceDirectoryName, destDirectoryName, StringComparison.OrdinalIgnoreCase);
```

Comment:
There is no need for `IsNullOrEmpty`, `string.Equals` handles that. Additionally, don't do this here, put it inline with the condition below. It isn't used until then which makes this really hard to follow.

Reply:
Fair. If you don't mind clarifying I'm doing this as I want to check if a directory with a case-sensitive version of that already exists in a different directory. So if I try to move `foo` to a directory with `FOO`, it wouldn't fail when I wanted it to. Do we only want to supper move as a renaming function? If so I can seriously reduce this complexity and make the contains not have lots of heap allocations for the strings.

Reply:
Why doesn't `FileSystem.DirectoryExists(fulldestDirName)` do that for you? If you're moving `/A/B` to `/B/b` when there is a `/B/B` or  `C:\A\B` to `C:\B\b` when there is a `C:\B\B` that would succeed/fail appropriately (given normal casing rules for Unix/Windows).

Reply:
This is an implementation question by me. Directory.Exists doesn't seem to check if a directory of a different case exists there. When I was debugging on W10, I moved `foo` into a folder with `fOo` . Maybe I'm misremembering but do we just want to let the OS see if the directory with the same case? If so that's a lot easier than my current scanning of all directories.

Reply:
Letting the OS check whether casing is a conflict is the best route to go. If the OS tells us it doesn't exist, then there is no conflict and we should let it through.